rules/type-error.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
+	'isFrozen',
+	'isFunction',
+	'isInteger',
+	'isLength',
+	'isMap',
+	'isNaN',
+	'isNative',
+	'isNil',
+	'isNull',
+	'isNumber',
+	'isObject',
+	'isObjectLike',
+	'isPlainObject',
+	'isPrototypeOf',
+	'isRegExp',
+	'isSafeInteger',
+	'isSealed',
+	'isSet',
+	'isString',
+	'isSymbol',
+	'isTypedArray',
+	'isUndefined',
+	'isView',
+	'isWeakMap',
+	'isWeakSet',
+	'isWindow',
+	'isXMLDoc'
+]);
+
+const isTypecheckingProvider = node => node.type === 'Identifier' && false;
+const isTypecheckingOperator = operator => operator === 'typeof' || operator === 'instanceof';
```

Comment:
Okay - i proved the `TypeCheckingOperator` function and for me it seems warrantable since its purpose is "just" to identify if this is a type checking operation. However, since this is adding a little performance penalty i removed it as suggested.. Thanks for the hint.. 🐎